label_studio/io_storages/azure_blob/models.py

```diff
@@ -125,7 +142,9 @@ def resolve_azure_blob(self, url):
 
         expiry = datetime.utcnow() + timedelta(minutes=self.presign_ttl)
 
-        sas_token = generate_blob_sas(account_name=self.get_account_name(),
+        sas_token = self.get_sas_token()
+        if not sas_token:
+            sas_token = generate_blob_sas(account_name=self.get_account_name(),
```

Comment:
Please add a debug-level logging message that differentiates the origin of SAS token, e.g:
```
if sas_token:
    logger.debug('Use specified SAS token')
else:
   logger.debug('Generate BLOB SAS token using Account Name and Account Key')
   ...
```